src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1643,6 +1659,7 @@ void goto_instrument_parse_optionst::help()
     " --nondet-static-exclude e    same as nondet-static except for the variable e\n" //NOLINT(*)
     "                              (use multiple times if required)\n"
     " --check-invariant function   instruments invariant checking function\n"
+    " --assert-ensures function    checks that function satisfies its contract ensures after each call\n" // NOLINT(*)
     " --remove-pointers            converts pointer arithmetic to base+offset expressions\n" // NOLINT(*)
```

Comment:
What does "after each call" mean?
So this is really checking the post-conditions in particular calling contexts?

Reply:
> Seems sensible overall -- perhaps call it `--check-postconditions`?

I agree that this is a much better name.


> What does "after each call" mean?
> So this is really checking the post-conditions in particular calling contexts?

It asserts the postcondition after all calls to the function. So it checks the postcondition if the function is reachable, and is called with its name (i.e. it wouldn't work for function pointers). If this is a problem, I could re-implement this as a check before each return point in the function body. I didn't do that initially for simplicity. What do you think?

Reply:
Or maybe `--check-ensures` (because they are written in the source code as `__CPROVER_ensures`) or `--check-contracts` ?

Reply:
What about `--check-contract-postconditions` or `--check-contract-ensures`?

Reply:
My personal opinion is that I like these `--check-blah` options to mirror whatever is in the source code - so I'd lean towards `--check-ensures` or `--check-contract-ensures` - others may have different opinions :-)